src/sentry/features/manager.py

```diff
@@ -222,29 +222,34 @@ def has(
         >>> FeatureManager.has('organizations:feature', organization, actor=request.user)
 
         """
-        try:
-            actor = kwargs.pop("actor", None)
-            feature = self.get(name, *args, **kwargs)
+        from sentry.db.models.manager.base import BaseManager
 
-            # Check registered feature handlers
-            rv = self._get_handler(feature, actor)
-            if rv is not None:
-                return rv
+        # Use thread local cache to store any cached models such as Subscriptions in addition to Django's cache.
+        # This helps shortcut around memcache's performance issues
+        with BaseManager.local_cache():
```

Comment:
This will help with de-duplicate queries within a single feature check. Will the cache be shared across multiple calls to `features.has()`?

Reply:
@markstory Sharing the cache across multiple `features.has()` complicates things.

I think caching within a single feature check is a good starting point since we don't expect to do any writes that necessitates invalidating the cache.

I've also added https://github.com/getsentry/sentry/pull/61787/commits/f9ada7cc22dafab0cb2ebaf58ee5e1cec93e7252 to flush the cache prior to `features.has()` returning.

For this profile of a single `features.has()` check that takes ~3.9 seconds to run, I've highlighted all occurrences of `DatabaseBackedSubscriptionService.get_for_organization` that we could potentially cache:

<img width="1712" alt="Screenshot 2023-12-15 at 3 23 42 PM" src="https://github.com/getsentry/sentry/assets/139499/ea49f99a-b7a1-4fd8-ae20-ee40f3166572">

Reply:
> For this profile of a single features.has() check that takes ~3.9 seconds to run, I've highlighted all occurrences of DatabaseBackedSubscriptionService.get_for_organization that we could potentially cache:

Wow that's a ton of queries :cry:. While those are local queries now, they'll eventually become RPC calls and a local ORM cache won't help as much.